llvm/tools/sycl-post-link/sycl-post-link.cpp

```diff
@@ -102,30 +103,41 @@ static void collectKernelsSet(
   }
 }
 
+// Input parameter KernelsSet is a map containing groups of kernels with same
+// values in the sycl-module-id attribute.
+// ResSymbolsLists vector is output parameter.
+// Collects set of kernel names for each group of kernels.
+static void collectSymbolsLists(
+    std::map<std::string, std::vector<Function *>> &KernelsSet,
+    std::vector<std::string> &ResSymbolsLists) {
+  for (auto &It : KernelsSet) {
+    std::string SymbolsList;
+    for (auto &F : It.second) {
```

Comment:
Technically it is a reduction algorithm such as `std::accumulate`...
Then why not just declaring `SymbolsList` to be a `Twine` to avoid the string conversions back-and-forth?

Reply:
Not sure. According to LLVM docs, `Twine` is a temporary object and shouldn't be stored. I use the `SymbolsList` next...
https://llvm.org/doxygen/classllvm_1_1Twine.html

Reply:
Agree with @Fznamznon. += can't be done on Twines, AFAIK.

Reply:
Ah yes, it looks like the `Twine` API is quite lower level. Sorry for the noise. :-(